### PR TITLE
Updated the max zoom level for grey basemaps

### PR DESCRIPTION
**Pull request by daraobeirne**
Hey updated the zoom level for the grey basemaps from 16 to 23. Let me know if there are issues or you have questions.

Thanks,
Dara

### CODE REVIEW

**Comment by steveoh** [comment]
refs #980

**Comment by jgravois** [comment]
thanks @daraobeirne! (and @steveoh!) either @jonahadkins or i will review this and help get it landed _very, very_ soon.

one of us will need to pull the branch down locally and make sure everything looks good somewhere with a lot of density like NYC and somewhere else more 'normal'.

if all is well, then we can update the other basemap layers so that they can be zoomed all the way down to `23` if they also support it.

**Comment by daraobeirne** [comment]
Thanks @jgravois  and @steveoh! I only updated the Grey basemaps and I noticed the world topo can go all the way up to 23 (insert Spinal Tap joke). So, if someone else still wanted to contribute that change could be made. Also, I was having issues on the "npm start" in my windows environment. Something to do with the rollup.js. I ended up doing it on my Linux box and it worked fine. If you're interested I can send along the log file later.

**Comment by jgravois** [comment]
@jonahadkins took a look at this pull request and was able to confirm several things.

1. despite the service metadata, our gray basemaps are *still* only cached down to level `16`.
2. other basemaps like our World Topo and World Street Map are *still* only cached to level `19`.
3. you **can** zoom in all the way to `23` in the arcgis.com map viewer, but this is only because tiles from `19` are resampled.

after reading the blog below, we realized that our World Imagery service in particular includes ultra-high resolution 5cm photography in limited areas, allowing for a native zoom of `22`.

https://blogs.esri.com/esri/arcgis/2017/04/25/world-imagery-updates-april-2017/

because of this, Jonah suggested that we consider updating the default maxZoom for that layer in this PR instead, but my preference would be *not* to do this so that we can avoid displaying blank tiles in areas where we don't have such high resolution aerial photography (see #240 for more info).

folks in Invercargill, New Zealand can still easily override the default to show off what their stuff, which is so good you can see [individual people](http://jsbin.com/wurekinuli/edit?html,output).

```js
var map = L.map('map').setView([-46.39802877, -191.64462067], 19);
L.esri.basemapLayer('Imagery', { maxZoom: 22 }).addTo(map);
```

tl;dr
it appears i've sent everyone on a ["goose chase"](http://www.urbandictionary.com/define.php?term=goose%20chase). for that i sincerely apologize. i'll find another opportunity to get @daraobeirne and the rest involved with something soon.

**Comment by daraobeirne** [comment]
Hey @jgravois no worries at all, it was fun to experiment in the first place and learned a little too. Maybe I can help out next time. Cheers!

**Comment by jgravois** [comment]
@daraobeirne Also, I was having issues on the "npm start" in my windows environment. Something to do with the rollup.js.

_please_ log this as a new issue whenever you have time. we definitely want to make sure windows folks don't have trouble getting up and running.
